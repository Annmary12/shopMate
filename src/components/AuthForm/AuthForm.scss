@import '../../assets/scss/variables.scss';

.auth-form {
  padding: 4rem;
  text-align: center;

  .facebook-login {
    display: flex;
    flex-direction: column;

   span:first-child {
      font-size: 2rem;
      padding-top: 1.5rem;
      font-family: 'Roboto';
      font-weight: 700;
    }
  }
  h1 {
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: bold;
    font-size: 2.4rem;
    line-height: 3.6rem;
    text-align: center;
    letter-spacing: .103125rem;

    color: #2E2E2E;
  }

  input {
    &::placeholder {
      font-family: 'Roboto';
      font-style: normal;
      font-weight: normal;
      font-size: 1.6rem;
      line-height: 2.4rem;
      text-align: center;

      color: #6C6C6C;

    }
  }

  button {
    margin-top: 2rem;
  }

  .remember-me {
    display: flex;
    justify-content: center;
    align-items: center;

    label {
      font-family: 'Montserrat';
      font-style: normal;
      font-weight: bold;
      font-size: 1.6rem;
      line-height: 2.4rem;
      color: #B4B4B4;
    }
  }

  .footer {
    padding: 0;
    padding-top: 4rem;
    font-family: 'Roboto';
    font-style: normal;
    font-weight: normal;
    font-size: 1.6rem;
    line-height: 2.4rem;
    color: $error-color;

    label {
      cursor: pointer;
    }
  }

  .signup-footer {
    padding-top: 2rem;
    font-family: 'Open Sans', sans-serif;
    font-style: normal;
    font-weight: 300;
    font-size: 1.6rem;
    line-height: 2.4rem;
    color: #2E2E2E;

    span {
      color: $error-color;
      padding-left: 1.5rem;
      cursor: pointer;
    }
  }
}