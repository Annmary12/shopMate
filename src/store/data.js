export const sizes = ['XS', 'S', 'M', 'L', 'XL', 'XXL'] ;
export const colors = [
  { name: 'blue' },
  { name: 'cyan' },
  { name: 'pink' },
  { name: 'orange' },
  { name: 'yellow' },
  { name: 'green' },
  { name: 'purple' }
];
