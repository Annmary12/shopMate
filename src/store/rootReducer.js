import { combineReducers } from 'redux';

// reducers
import products from './modules/products';
import cart from './modules/cart';
import auth from './modules/auth';
import customer from './modules/customer';
import regions from './modules/shippingOptions';
import orders from './modules/order';
import categories from './modules/categories';
import department from './modules/department';

const appReducer = combineReducers({
  department,
  categories,
  products,
  cart,
  auth,
  customer,
  regions,
  orders
});

const rootReducer = (state, action) => {
  return appReducer(state, action);
};

export default rootReducer;