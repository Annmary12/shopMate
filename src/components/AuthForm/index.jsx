
import React, { Component } from 'react';

// third party libraries
import PropTypes from 'prop-types';
import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import _ from 'lodash';
import FacebookLogin from 'react-facebook-login';

// components
import InputBox from '../InputBox';
import Button from '../Button';
import CheckBox from '../CheckBox';

// validation schema
import { SignUpSchema, LoginSchema } from '../../utils/validationSchemas/auth';

import { registerUser, login, loginWithFacebook } from 'store/modules/auth'

// style
import './AuthForm.scss';


export class AuthForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      hidden: true
    }
  }

  /**
   *  Gets the submit button text
   *
   * @returns {string}
   */
  SubmitButtonText = () => {
    switch (this.props.formType) {
      case 'signup':
        return 'Sign Up';
      case 'login':
        return 'Sign In';
      case 'forgotPassword':
        return 'Send Password Reset Link';
      default:
        return 'Sign Up';
    }
  }

  /**
   * Checks if is a signup form
   *
   * @returns {boolean}
   */
  isSignUp = () => this.props.formType === 'signup';

  /**
   * Checks if is a login form
   *
   * @returns {boolean}
   */
  isLogin = () => this.props.formType === 'login';

  /**
   * checks if is a password reset form
   *
   * @returns {boolean}
   */
  isPasswordReset = () => (this.props.formType === 'forgotPassword');

  responseFacebook = (response) => {
    if (response.accessToken) {
      this.props.loginWithFacebook(response.accessToken, this.props.closeModal);
    }
  }

  render() {
    const {
      values,
      touched,
      errors,
      isSubmitting,
      handleChange,
      handleSubmit,
      formType
    } = this.props;

    return (
      <div className="auth-form">
        { this.props.hasError &&
          <div className="error-message">
              {this.props.hasError}
          </div>
        }
        <h1>{ formType == 'login' ? 'Sign In' : 'Sign Up'}</h1>
        <form onSubmit={handleSubmit} className="form">
          {
            this.isSignUp() &&
            <InputBox
              type="text"
              name="name"
              placeholder="Name"
              inputType="blocked"
              {...this.props}
            />
          }
          <InputBox
            type="email"
            name="email"
            placeholder="Email"
            inputType="blocked"
            {...this.props}
          />

          <InputBox
            type="password"
            name="password"
            inputType="blocked"
            placeholder="Password"
            {...this.props}
          />

          {
            this.isLogin() &&
            <div className="remember-me mbxs">
              <CheckBox
                name="rememberMe"
                {...this.props}
              /> <label htmlFor="remember">Remember</label>
            </div>
          }
          <Button
            classes="btn__primary"
            type="submit"
            size="medium"
            name={isSubmitting ? 'loading..' : this.SubmitButtonText()}
            disabled={isSubmitting}
          />
          {
            this.isLogin() &&
              <div className="facebook-login">
                <span>or</span>
                <FacebookLogin
                  appId="352854622106208"
                  fields="name,email,picture"
                  callback={this.responseFacebook}
                  icon="fa-facebook"
                />
              </div>
         }

          {
            this.isLogin() &&
            <div className="footer">
              <label htmlFor="forgotPassword">Forgot password</label>
              <label htmlFor="account" onClick={this.props.onChangeFormState('isRegister')}>Have an account</label>
            </div>
          }

          {
            this.isSignUp() && 
            <div className="signup-footer">
              <p>
                Already a member? <span onClick={this.props.onChangeFormState('isLogin')}>Sign in</span>
              </p>
            </div>
          }
        </form>

      </div>
    );
  }
}

const MyEnhancedAuthForm  = {
  mapPropsToValues: () => ({
    name: '',
    email: '',
    password: '',
    rememberMe: ''
  }),

  validationSchema: props => props.formType === 'signup' ? SignUpSchema : LoginSchema,

  handleSubmit: (values, { setSubmitting, props }) => {
     props.formType === 'signup'
      ? props.registerUser(values, props.closeModal)
      : props.login(values,  props.closeModal);
      setSubmitting(false);
  },
}

AuthForm.propTypes = {
  formType: PropTypes.string,
  values: PropTypes.objectOf(PropTypes.string),
  touched: PropTypes.objectOf(PropTypes.bool),
  errors: PropTypes.objectOf(PropTypes.string),
  isSubmitting: PropTypes.bool,
  handleChange: PropTypes.func.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  onChangeFormState: PropTypes.func
};

AuthForm.defaultProps = {
  formType: '',
  values: {},
  touched: {},
  errors: {},
  isSubmitting: false
};

export const mapStateToProps = state => {
  return {
    hasError: state.auth.error,
    isLoading: state.auth.isLoading,
    authUser: state.auth.data
  }
};

export const mapDispatchToProps = dispatch => ({
  registerUser: (userDetails, closeModal) => dispatch(registerUser(userDetails, closeModal)),
  login: (userDetails, closeModal) => dispatch(login(userDetails, closeModal)),
  loginWithFacebook: (accessToken, closeModal) => dispatch(loginWithFacebook(accessToken, closeModal)),
})

export default withRouter(compose(connect(mapStateToProps, mapDispatchToProps), withFormik(MyEnhancedAuthForm))(AuthForm));