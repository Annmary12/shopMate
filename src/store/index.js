import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from  'redux-devtools-extension';
import reduxImmutableStateInvariant from 'redux-immutable-state-invariant';

import thunk from 'redux-thunk';

import http from '../utils/helpers/http';

import rootReducer from './rootReducer';

const devMiddleWare = composeWithDevTools(
  applyMiddleware(
    thunk.withExtraArgument(http),
    reduxImmutableStateInvariant()
  )
);
const prodMiddleware = applyMiddleware(thunk.withExtraArgument(http));

const middleware = process.env.NODE_ENV === 'development' ? devMiddleWare : prodMiddleware;

const store = createStore(
  rootReducer,
  {},
  middleware
)

export default store;