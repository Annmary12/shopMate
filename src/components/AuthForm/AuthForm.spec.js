// react library
import React from 'react';

// third-party library
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../store';

// component
import AuthForm from './';


describe('AuthForm component', () => {
  let wrapper;
  const props = {
    formType: 'login',
    onChangeFormState: jest.fn()
  }

  // beforeEach(() => {
  //  wrapper = shallow(
  //   // <Provider store={store}>
  //     {/* <MemoryRouter> */}
  //       <AuthForm { ...props } />
  //     {/* </MemoryRouter> */}
  //   // </Provider>
  //   )
  // })
  it('should render correctly', () => {
    wrapper = shallow(<AuthForm.WrappedComponent { ...props } />);
    // console.log(wrapper.debug());
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})