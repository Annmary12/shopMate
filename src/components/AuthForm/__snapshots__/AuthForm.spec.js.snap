// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthForm component should render button correctly 1`] = `undefined`;

exports[`AuthForm component should render correctly 1`] = `
<Formik
  enableReinitialize={false}
  formType="login"
  handleSubmit={[Function]}
  initialValues={
    Object {
      "email": "",
      "name": "",
      "password": "",
      "rememberMe": "",
    }
  }
  isInitialValid={false}
  onChangeFormState={[MockFunction]}
  onSubmit={[Function]}
  render={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
  validationSchema={[Function]}
/>
`;
